warning CS8618 reported for uninitialized field marked [MaybeNull]

```C#
#nullable enable
#pragma warning disable 169
#pragma warning disable 414

using System.Diagnostics.CodeAnalysis;

class C<T>
{
    [MaybeNull] T F1;
    [MaybeNull] object F2;
}
```
```
(9,19): warning CS8618: Non-nullable field 'F1' is uninitialized.
(10,24): warning CS8618: Non-nullable field 'F2' is uninitialized.
```

--- a/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs
+++ b/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs
@@ -4251,15 +4251,7 @@ namespace Microsoft.CodeAnalysis.CSharp
                 return FlowAnalysisAnnotations.None;
             }
 
-            var annotations = symbol switch
-            {
-                MethodSymbol method => method.ReturnTypeFlowAnalysisAnnotations,
-                PropertySymbol property => property.GetOwnOrInheritedGetMethod()?.ReturnTypeFlowAnalysisAnnotations ?? FlowAnalysisAnnotations.None,
-                ParameterSymbol parameter => parameter.FlowAnalysisAnnotations,
-                FieldSymbol field => field.FlowAnalysisAnnotations,
-                _ => FlowAnalysisAnnotations.None
-            };
-
+            var annotations = symbol.GetFlowAnalysisAnnotations();
             return annotations & (FlowAnalysisAnnotations.MaybeNull | FlowAnalysisAnnotations.NotNull);
         }
 

--- a/src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs
+++ b/src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs
@@ -259,6 +259,11 @@ namespace Microsoft.CodeAnalysis.CSharp
                     default:
                         continue;
                 }
+                var annotations = (field.IsImplicitlyDeclared ? field.AssociatedSymbol : field).GetFlowAnalysisAnnotations();
+                if ((annotations & FlowAnalysisAnnotations.MaybeNull) != 0)
+                {
+                    continue;
+                }
                 if (field.IsConst)
                 {
                     continue;

--- a/src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs
+++ b/src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs
@@ -259,11 +259,6 @@ namespace Microsoft.CodeAnalysis.CSharp
                     default:
                         continue;
                 }
-                var annotations = (field.IsImplicitlyDeclared ? field.AssociatedSymbol : field).GetFlowAnalysisAnnotations();
-                if ((annotations & FlowAnalysisAnnotations.MaybeNull) != 0)
-                {
-                    continue;
-                }
                 if (field.IsConst)
                 {
                     continue;
@@ -285,6 +280,10 @@ namespace Microsoft.CodeAnalysis.CSharp
                     FieldSymbol { AssociatedSymbol: PropertySymbol p } => p,
                     _ => member
                 };
+                if ((symbol.GetFlowAnalysisAnnotations() & FlowAnalysisAnnotations.MaybeNull) != 0)
+                {
+                    continue;
+                }
                 var location = getSymbolForLocation(walkerOpt, symbol).Locations.FirstOrNone();
                 diagnostics.Add(ErrorCode.WRN_UninitializedNonNullableField, location, symbol.Kind.Localize(), symbol.Name);
             }

--- a/src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs
@@ -419,6 +419,18 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             return returnType;
         }
 
+        internal static FlowAnalysisAnnotations GetFlowAnalysisAnnotations(this Symbol? symbol)
+        {
+            return symbol switch
+            {
+                MethodSymbol method => method.ReturnTypeFlowAnalysisAnnotations,
+                PropertySymbol property => property.GetOwnOrInheritedGetMethod()?.ReturnTypeFlowAnalysisAnnotations ?? FlowAnalysisAnnotations.None,
+                ParameterSymbol parameter => parameter.FlowAnalysisAnnotations,
+                FieldSymbol field => field.FlowAnalysisAnnotations,
+                _ => FlowAnalysisAnnotations.None
+            };
+        }
+
         internal static void GetTypeOrReturnType(this Symbol symbol, out RefKind refKind, out TypeWithAnnotations returnType,
                                                  out ImmutableArray<CustomModifier> refCustomModifiers)
         {

--- a/src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs
@@ -419,7 +419,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             return returnType;
         }
 
-        internal static FlowAnalysisAnnotations GetFlowAnalysisAnnotations(this Symbol? symbol)
+        internal static FlowAnalysisAnnotations GetFlowAnalysisAnnotations(this Symbol symbol)
         {
             return symbol switch
             {

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs
@@ -1379,12 +1379,13 @@ class C
         }
 
         [Fact]
-        public void MaybeNullT_Property_Uninitialized()
+        public void MaybeNullT_Uninitialized()
         {
             var source =
 @"using System.Diagnostics.CodeAnalysis;
 public class C<T>
 {
+    [MaybeNull] public T F;
     [MaybeNull] public T P { get; set; }
 }";
             var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
@@ -1393,26 +1394,13 @@ public class C<T>
         }
 
         [Fact]
-        public void MaybeNullT_Field_Uninitialized()
-        {
-            var source =
-@"using System.Diagnostics.CodeAnalysis;
-public class C<T>
-{
-    [MaybeNull] public T P;
-}";
-            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
-            comp.VerifyDiagnostics(
-                );
-        }
-
-        [Fact]
-        public void MaybeNull_ClassT_Property_Uninitialized()
+        public void MaybeNull_ClassT_Uninitialized()
         {
             var source =
 @"using System.Diagnostics.CodeAnalysis;
 public class C<T> where T : class
 {
+    [MaybeNull] public T F;
     [MaybeNull] public T P { get; set; }
 }";
             var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
@@ -1421,13 +1409,14 @@ public class C<T> where T : class
         }
 
         [Fact]
-        public void MaybeNull_ClassT_Field_Uninitialized()
+        public void MaybeNull_NotNullT_Uninitialized()
         {
             var source =
 @"using System.Diagnostics.CodeAnalysis;
-public class C<T> where T : class
+public class C<T> where T : notnull
 {
-    [MaybeNull] public T P;
+    [MaybeNull] public T F;
+    [MaybeNull] public T P { get; set; }
 }";
             var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics(
@@ -1435,12 +1424,13 @@ public class C<T> where T : class
         }
 
         [Fact]
-        public void MaybeNull_Property_Uninitialized()
+        public void MaybeNull_Uninitialized()
         {
             var source =
 @"using System.Diagnostics.CodeAnalysis;
 public class C
 {
+    [MaybeNull] public string F;
     [MaybeNull] public string P { get; set; }
 }";
             var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
@@ -1449,50 +1439,23 @@ public class C
         }
 
         [Fact]
-        public void MaybeNull_Field_Uninitialized()
+        public void MaybeNull_NullInitializer()
         {
             var source =
 @"using System.Diagnostics.CodeAnalysis;
 public class C
 {
-    [MaybeNull] public string P;
-}";
-            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
-            comp.VerifyDiagnostics(
-                );
-        }
-
-        [Fact]
-        public void MaybeNull_Property_NullInitializer()
-        {
-            var source =
-@"using System.Diagnostics.CodeAnalysis;
-public class C
-{
+    [MaybeNull] public string F = null;
     [MaybeNull] public string P { get; set; } = null;
 }";
             var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics(
-                // (4,49): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //     [MaybeNull] public string P { get; set; } = null;
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 49)
-                );
-        }
-
-        [Fact]
-        public void MaybeNull_Field_NullInitializer()
-        {
-            var source =
-@"using System.Diagnostics.CodeAnalysis;
-public class C
-{
-    [MaybeNull] public string P = null;
-}";
-            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
-            comp.VerifyDiagnostics(
                 // (4,35): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //     [MaybeNull] public string P = null;
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 35)
+                //     [MaybeNull] public string F = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 35),
+                // (5,49): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     [MaybeNull] public string P { get; set; } = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(5, 49)
                 );
         }
     }

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs
@@ -523,9 +523,6 @@ class C<T, U, V>
                 // (7,22): warning CS8618: Non-nullable property 'P1' is uninitialized. Consider declaring the property as nullable.
                 //     private static T P1 { get; }
                 Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "P1").WithArguments("property", "P1").WithLocation(7, 22),
-                // (9,34): warning CS8618: Non-nullable property 'P3' is uninitialized. Consider declaring the property as nullable.
-                //     [MaybeNull] private static T P3 { get; }
-                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "P3").WithArguments("property", "P3").WithLocation(9, 34),
                 // (10,48): warning CS8601: Possible null reference assignment.
                 //     [MaybeNull] private static T P4 { get; } = default;
                 Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(10, 48),
@@ -844,9 +841,6 @@ class C<T>
                 // (12,13): warning CS8618: Non-nullable field 'F2' is uninitialized. Consider declaring the field as nullable.
                 //     private C(T t)
                 Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F2").WithLocation(12, 13),
-                // (12,13): warning CS8618: Non-nullable field 'F3' is uninitialized. Consider declaring the field as nullable.
-                //     private C(T t)
-                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F3").WithLocation(12, 13),
                 // (12,13): warning CS8618: Non-nullable property 'P2' is uninitialized. Consider declaring the property as nullable.
                 //     private C(T t)
                 Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("property", "P2").WithLocation(12, 13),
@@ -1382,6 +1376,124 @@ class C
                 //         P2 += "";
                 Diagnostic(ErrorCode.ERR_PropertyLacksGet, "P2").WithArguments("C.P2").WithLocation(12, 9)
             );
+        }
+
+        [Fact]
+        public void MaybeNullT_Property_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T>
+{
+    [MaybeNull] public T P { get; set; }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNullT_Field_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T>
+{
+    [MaybeNull] public T P;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_ClassT_Property_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T> where T : class
+{
+    [MaybeNull] public T P { get; set; }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_ClassT_Field_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C<T> where T : class
+{
+    [MaybeNull] public T P;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Property_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P { get; set; }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Field_Uninitialized()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Property_NullInitializer()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P { get; set; } = null;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (4,49): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     [MaybeNull] public string P { get; set; } = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 49)
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Field_NullInitializer()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [MaybeNull] public string P = null;
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (4,35): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     [MaybeNull] public string P = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 35)
+                );
         }
     }
 }

